src/swsssdk/dbconnector.py

```diff
@@ -9,182 +9,324 @@
 # FIXME: Convert to metaclasses when Py2 support is removed. Metaclasses have unique interfaces to Python2/Python3.
 
 class SonicDBConfig(object):
```

Comment:
>SonicDBConfig [](start = 6, length = 13)

You may define a new class (SonicNamespaceConfig) above original SonicDBConfig. The original class fits well in this layered design.

Reply:
I find from a bit of testing that it is easier to extract/keep the namespace details in SonicDBConfig itself, since we have the "namespace information" also in the database json files which we have to parse in the SonicDBConfig class.

Additionally this will help keep the entry point same as SonicDBConfig class irrespective of whether the SonicV2Connector/ConfigDBConnector is used with a "specific namespace" or not.